mmcls/models/backbones/convvit.py

```diff
@@ -0,0 +1,381 @@
+        assert num_stages == len(patch_size), \
+            'patch_size has to be specified for each stage exactly once.'
+        assert num_stages == len(num_layers), \
+            'num_layers has to be specified for each stage exactly once.'
+        assert num_stages == len(feedforward_channels), \
+            'feedforward_channels has to be specified for each stage ' \
+            'exactly once.'
+
+        dpr = np.linspace(0, drop_path_rate, sum(num_layers))
+
+        self.patch_embeds = nn.ModuleList([])
+        self.blocks = nn.ModuleList([])
+        self.patch_embed_act = build_activation_layer(act_cfg)
+
+        stage_img_size = img_size
+        if isinstance(layer_cfgs, dict):
+            layer_cfgs = [layer_cfgs] * sum(num_layers)
+        for stage_idx in range(num_stages):
+            _patch_cfg = dict(
+                in_channels=(
+                    in_channels if stage_idx == 0 else embed_dims[stage_idx - 1]
+                ),
+                input_size=stage_img_size,
+                embed_dims=embed_dims[stage_idx],
+                conv_type='Conv2d',
+                kernel_size=patch_size[stage_idx],
+                stride=patch_size[stage_idx],
+                norm_cfg=tx_norm_cfg,
+            )
+            _patch_cfg.update(patch_cfg)
+            patch_embed = PatchEmbed(**_patch_cfg)
+            self.patch_embeds.append(patch_embed)
+            stage_img_size = patch_embed.init_out_size
+
+            if stage_idx == 2:
+                self.patch_embeds.append(
+                    nn.Linear(embed_dims[stage_idx], embed_dims[stage_idx]))
+
+
+            blocks = []
+            for layer_idx in range(num_layers[stage_idx]):
+                global_layer_idx = sum(num_layers[:stage_idx]) + layer_idx
+                _layer_cfg = dict(
+                    embed_dims=embed_dims[stage_idx],
+                    feedforward_channels=feedforward_channels[stage_idx],
+                    drop_rate=drop_rate,
+                    drop_path_rate=dpr[global_layer_idx],
+                    act_cfg=act_cfg)
+                if stage_idx >= 2: # transformer stage
+                    layer_class = TransformerEncoderLayer
+                    _layer_cfg.update(
+                        num_heads=num_heads,
+                        qkv_bias=qkv_bias,
+                        norm_cfg=tx_norm_cfg,
+                    )
+                    self.pos_embed = nn.Parameter(
+                        torch.zeros(1, *stage_img_size, embed_dims[stage_idx]))
+                else: # conv stage
+                    layer_class = ConvBlock
+                    layer_conv_norm_cfg = conv_norm_cfg.copy()
+                    _layer_cfg.update(
+                        norm_cfg=layer_conv_norm_cfg,
+                        )
+                _layer_cfg.update(layer_cfgs[global_layer_idx])
+                blocks.append(layer_class(**_layer_cfg))
+            self.blocks.append(nn.ModuleList(blocks))
```

Comment:
Please use ``self.layers``, to be consistent with ``VisionTransformer`` in MMCls.